### PR TITLE
Introduce documentation for using GameLift Anywhere

### PR DESCRIPTION
**Pull request by AMZN-daimini**
Note that this documentation will only work with GameLift C++ SDK v5, so it should only be merged once that update has been completed.

### CODE REVIEW

File: MPSGameLift/Documentation/GameLift.md [hunk 1]
```diff
@@ -38,6 +38,16 @@ This README covers optional setup, testing and running on [Amazon GameLift](http
     ---
 It's important to make sure that the bootstrap.game.profile.setreg file has been added to one of the seed files. (also add debug if you want to support debug builds)
 
+---
+**NOTE**
+
+This will not build the game client. If you need the client for testing, also run this command:
+```
+cmake --build build\windows_mono --target MultiplayerSample.GameLauncher --config profile -- /m /nologo
```
Comment (lmbr-pip) [inline MPSGameLift/Documentation/GameLift.md:46]: I would add context that the export_gamelift_server_package sets up to build things monolithic etc

Also could we add this as an option ie -build-test-client = true? to that script so it just does it?

File: MPSGameLift/Documentation/GameLift.md [hunk 2]
```diff
@@ -81,7 +91,114 @@ It's important to make sure that the bootstrap.game.profile.setreg file has been
...
+C:\GameLiftPackageWindows\MultiplayerSample.ServerLauncher.exe --rhi=null -NullRenderer -bg_ConnectToAssetProcessor=0 --console-command-file=launch_server.cfg --sv_dedicated_host_onstartup=false --sv_gameLiftEnabled=true --sv_gameliftAnywhereEnabled=true --sv_gameliftAnywhereWebSocketUrl="<WebSocketUrl>" --sv_gameliftAnywhereAuthToken="<AuthToken>" --sv_gameliftAnywhereFleetId="<FleetId>" --sv_gameliftAnywhereHostId="<ComputeName>" --sv_gameliftAnywhereProcessId="<ProcessId>"
+```
+
+### Create a Game Session
+
+```sh
+aws gamelift create-game-session --region <Region> --location custom-location-1 --fleet-id <FleetId> --name GameSession1 --maximum-player-session-count 3
+```
+
+If the operation was successful, the console will display the json result.
```
Comment (lmbr-pip) [inline MPSGameLift/Documentation/GameLift.md:165]:
```suggestion
If the operation was successful, the console will display the JSON result.
```

File: MPSGameLift/Documentation/GameLift.md [hunk 3]
```diff
@@ -81,7 +91,114 @@ It's important to make sure that the bootstrap.game.profile.setreg file has been
...
+Record the `GameSessionId` for the next steps. Example: arn:aws:gamelift:us-west-2::gamesession/fleet-1a23bc4d-456e-78fg-h9i0-jk1l23456789/custom-location-1/gsess-ab1cd2ef-3gh4-5678-ijk9-0l1mn2o345p6
+
+If the operation fails, make sure the server is running and the `InitSDK` call was successful.
```
Comment (lmbr-pip) [inline MPSGameLift/Documentation/GameLift.md:168]:
```suggestion
If the operation fails, make sure the server is running. Ensure that `InitSDK` and `ProcessReady` calls were successful.
```

**Review by lmbr-pip — APPROVED** [review]

**Comment by AMZN-Gene** [comment]
Related PR https://github.com/aws-lumberyard-dev/o3de/pull/533

**Review by AMZN-Gene — APPROVED** [review]

File: MPSGameLift/Documentation/GameLift.md
[the remark this answers is not in the document]
Comment (AMZN-daimini) [reply MPSGameLift/Documentation/GameLift.md:1]: I originally went for a separate file as you suggest, but over 50% of the steps would have been duplicated. So I would've either had to ask the reader to reference the other file and go back and forth, or copy a good chunk of the content and made it harder to maintain.

Honestly I'm fine with either option. If we want to keep all in one doc, I will definitely add a bit more of an intro to explain the differences.

File: MPSGameLift/Documentation/GameLift.md (hunk 1 above)
Comment (AMZN-daimini) [reply MPSGameLift/Documentation/GameLift.md:46]: Yes, we discussed having an option with Gene. If we want to do that, we can spin up a separate task for it and either of us can pick it up.

File: MPSGameLift/Documentation/GameLift.md (hunk 2 above)
Comment (AMZN-daimini) [reply MPSGameLift/Documentation/GameLift.md:165]: Fixed everywhere!

**Review by cgalvan — APPROVED** [review]

File: MPSGameLift/Documentation/GameLift.md [hunk 4]
```diff
@@ -1,8 +1,8 @@
 # MultiplayerSample Project with Amazon GameLift
 
-This README covers optional setup, testing and running on [Amazon GameLift](https://aws.amazon.com/gamelift/), an AWS service to make hosting and scaling game servers easier. 
+This README covers optional setup, testing and running on [Amazon GameLift](https://aws.amazon.com/gamelift/), an AWS service to make hosting and scaling game servers easier. It also provides guidance on how to test the Amazon GameLift integration on your local machine via GameLift Anywhere.
```
Comment (allisaurus) [inline MPSGameLift/Documentation/GameLift.md:3]:
```suggestion
This README covers optional setup, testing and running on [Amazon GameLift](https://aws.amazon.com/gamelift/), an AWS service to make hosting and scaling game servers easier. It also provides guidance on how to test the Amazon GameLift integration on your local machine via [Amazon GameLift Anywhere](https://docs.aws.amazon.com/gamelift/latest/developerguide/fleets-creating-anywhere.html).
```

File: MPSGameLift/Documentation/GameLift.md [hunk 5]
```diff
@@ -30,13 +30,18 @@ This README covers optional setup, testing and running on [Amazon GameLift](http
 
     ---
     
-    ---
-    **Important**
+It's important to make sure that the bootstrap.game.profile.setreg file has been added to one of the seed files. (also add debug if you want to support debug builds)
```
Comment (allisaurus) [inline MPSGameLift/Documentation/GameLift.md:33]:
```suggestion
It's important to make sure that the _bootstrap.game.profile.setreg_ file has been added to one of the seed files. (also add debug if you want to support debug builds)
```

File: MPSGameLift/Documentation/GameLift.md [hunk 6]
```diff
@@ -30,13 +30,18 @@ This README covers optional setup, testing and running on [Amazon GameLift](http
...
 
-    The export_gamelift_server_package script only works for projects built using engine source, and won't work with engine as an sdk. 
+---
+**NOTE**
 
-    ---
-It's important to make sure that the bootstrap.game.profile.setreg file has been added to one of the seed files. (also add debug if you want to support debug builds)
+This will not build the game client. If you need the client for testing, also run this command:
+```
```
Comment (allisaurus) [inline MPSGameLift/Documentation/GameLift.md:39]:
```suggestion
```sh
```
to improve syntax highlighting

File: MPSGameLift/Documentation/GameLift.md [hunk 7]
```diff
@@ -81,7 +86,113 @@ It's important to make sure that the bootstrap.game.profile.setreg file has been
...
+### Create a fleet
+
+Creating an Anywhere fleet is a much faster process compared to creating a regular AWS fleet, which usually takes about an hour to setup.
+
+```sh
+aws gamelift create-fleet --name AnywhereFleet --compute-type ANYWHERE --locations Location=custom-location-1 --region <Region>
+```
+
+If the operation was successful, the console will display the JSON result.
+Record the `FleetId` for the next steps. Example: fleet-1a23bc4d-456e-78fg-h9i0-jk1l23456789
```
Comment (allisaurus) [inline MPSGameLift/Documentation/GameLift.md:116]:
```suggestion
Record the `FleetId` for the next steps. Example: **fleet-1a23bc4d-456e-78fg-h9i0-jk1l23456789**
```

File: MPSGameLift/Documentation/GameLift.md [hunk 8]
```diff
@@ -81,7 +86,113 @@ It's important to make sure that the bootstrap.game.profile.setreg file has been
...
+
+### Register your local machina as a Compute
```
Comment (allisaurus) [inline MPSGameLift/Documentation/GameLift.md:118]:
```suggestion
### Register your local machine as a Compute
```

File: MPSGameLift/Documentation/GameLift.md [hunk 9]
```diff
@@ -81,7 +86,113 @@ It's important to make sure that the bootstrap.game.profile.setreg file has been
...
+Register your local machine as a GameLift Anywhere Compute.
+For ease of testing, we assume the Server and Client will be run on the same machine; so we can pass localhost (`127.0.0.1`) as the IP address.
+If your machine is accessible via a public IP address, change that value as appropriate.
+
+```sh
+aws gamelift register-compute --compute-name CustomCompute1 --fleet-id <FleetId> --ip-address 127.0.0.1 --location custom-location-1 --region <Region>
+```
+
+If the operation was successful, the console will display the JSON result.
+Record the `ComputeName` for the next steps. Example: CustomCompute1
```
Comment (allisaurus) [inline MPSGameLift/Documentation/GameLift.md:129]:
```suggestion
Record the `ComputeName` for the next steps. Example: **CustomCompute1**
```

File: MPSGameLift/Documentation/GameLift.md [hunk 10]
```diff
@@ -81,7 +86,113 @@ It's important to make sure that the bootstrap.game.profile.setreg file has been
...
+Also record the `GameLiftServiceSdkEndpoint` for later. Example: wss://us-west-2.api.amazongamelift.com
```
Comment (allisaurus) [inline MPSGameLift/Documentation/GameLift.md:130]:
```suggestion
Also record the `GameLiftServiceSdkEndpoint` for later. Example: **wss://us-west-2.api.amazongamelift.com**
```

File: MPSGameLift/Documentation/GameLift.md [hunk 11]
```diff
@@ -81,7 +86,113 @@ It's important to make sure that the bootstrap.game.profile.setreg file has been
...
+
+### Get Compute auth token
+
+```sh
+aws gamelift get-compute-auth-token --fleet-id <FleedId> --compute-name <ComputeName>
+```
+
+If the operation was successful, the console will display the JSON result.
+Record the `AuthToken` for the next steps. Example: 123a4b5c-d6e7-8fgh-9i01-2jklm34no567
```
Comment (allisaurus) [inline MPSGameLift/Documentation/GameLift.md:139]:
```suggestion
Record the `AuthToken` for the next steps. Example: **123a4b5c-d6e7-8fgh-9i01-2jklm34no567**
```

File: MPSGameLift/Documentation/GameLift.md [hunk 12]
```diff
@@ -81,7 +86,113 @@ It's important to make sure that the bootstrap.game.profile.setreg file has been
...
+```
+
+### Create a Game Session
+
+```sh
+aws gamelift create-game-session --region <Region> --location custom-location-1 --fleet-id <FleetId> --name GameSession1 --maximum-player-session-count 3
+```
+
+If the operation was successful, the console will display the JSON result.
+Record the `GameSessionId` for the next steps. Example: arn:aws:gamelift:us-west-2::gamesession/fleet-1a23bc4d-456e-78fg-h9i0-jk1l23456789/custom-location-1/gsess-ab1cd2ef-3gh4-5678-ijk9-0l1mn2o345p6
```
Comment (allisaurus) [inline MPSGameLift/Documentation/GameLift.md:160]:
```suggestion
Record the `GameSessionId` for the next steps. Example: **arn:aws:gamelift:us-west-2::gamesession/fleet-1a23bc4d-456e-78fg-h9i0-jk1l23456789/custom-location-1/gsess-ab1cd2ef-3gh4-5678-ijk9-0l1mn2o345p6**
```

File: MPSGameLift/Documentation/GameLift.md [hunk 13]
```diff
@@ -81,7 +86,113 @@ It's important to make sure that the bootstrap.game.profile.setreg file has been
...
+### Create a Player Session
+
+Note: `PlayerId` from the JSON above and `--player-id` in the command below do not need to be the same
+
+```sh
+aws gamelift create-player-session --region <Region> --game-session-id <GameSessionId> --player-id Player1
+```
+
+If the operation was successful, the console will display the JSON result.
+Record the `PlayerSessionId` for the next steps. Example: psess-1a2b3c45-d6e7-89fg-0hij-12kl34m56no7
```
Comment (allisaurus) [inline MPSGameLift/Documentation/GameLift.md:186]:
```suggestion
Record the `PlayerSessionId` for the next steps. Example: **psess-1a2b3c45-d6e7-89fg-0hij-12kl34m56no7**
```

File: MPSGameLift/Documentation/GameLift.md [hunk 14]
```diff
@@ -94,7 +205,7 @@ Builds are tied to Fleets; you may want to delete the existing build and fleet v
 ```sh
 aws gamelift upload-build --operating-system WINDOWS_2016 --build-root C:\GameLiftPackageWindows\ --name MultiplayerSample --build-version v1.0 --region us-west-2
 ```
-Record BuildId for the next step. Example: build-fe4a2b2c-8ae5-4757-99a6-0ff372d03512
+Record BuildId for the next step. Example: build-1a23bc4d-456e-78fg-h9i0-jk1l23456789
```
Comment (allisaurus) [inline MPSGameLift/Documentation/GameLift.md:208]:
```suggestion
Record BuildId for the next step. Example: **build-1a23bc4d-456e-78fg-h9i0-jk1l23456789**
```

File: MPSGameLift/Documentation/GameLift.md [hunk 15]
```diff
@@ -109,7 +220,7 @@ The ec2-instance-type and fleet-type determines the kind of AWS resources used;
 
 ---
 
-Record the FleetId for the next step. Example: fleet-1a49fc3e-892a-40fc-b2e9-aa7e11983182
+Record the FleetId for the next step. Example: fleet-1a23bc4d-456e-78fg-h9i0-jk1l23456789
```
Comment (allisaurus) [inline MPSGameLift/Documentation/GameLift.md:223]:
```suggestion
Record the FleetId for the next step. Example: **fleet-1a23bc4d-456e-78fg-h9i0-jk1l23456789**
```

**Review by allisaurus — APPROVED** [review]

**Pull request merged**